cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -321,6 +324,34 @@ func gitIndex(c gitIndexConfig, o *indexArgs, l sglog.Logger) error {
 		"-submodules=false",
 	}
 
+	// We store the document ranks as JSON in gitDir and add a flag to tell
+	// zoekt-git-index to rank based on the scores in the file, instead of ranking
+	// the files by itself.
+	if o.UseOfflineRanking {
+		args = append(args, "-offline_ranking")
+
+		r, err := sourcegraph.GetDocumentRanks(context.Background(), o.Name)
+		if err != nil {
+			return err
+		}
+
+		b, err := json.Marshal(r)
```

Comment:
design: given you are keeping these arguments in memory, instead of passing in a sourcegraph interface, could you not specify the ranking stuff as part of `indexArgs` rather than as a boolean?

Later: I can imagine you writing this stuff straight to disk and avoiding storing it in memory, in that case as a boolean makes sense.